java-source/src/test/java/net/corda/training/contract/IOUSettleTests.java

```diff
@@ -1,5 +1,524 @@
 package net.corda.training.contract;
 
-public class IOUSettleTests{
-	
+import net.corda.core.contracts.*;
+import net.corda.core.contracts.Amount;
+import net.corda.core.contracts.CommandData;
+import net.corda.core.contracts.PartyAndReference;
+import net.corda.core.contracts.TypeOnlyCommandData;
+import net.corda.core.identity.AbstractParty;
+import net.corda.core.utilities.OpaqueBytes;
+import net.corda.finance.Currencies;
+import net.corda.finance.contracts.asset.Cash;
+import net.corda.testing.node.MockServices;
+import net.corda.training.state.IOUState;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Currency;
+
+import static net.corda.testing.node.NodeTestUtils.ledger;
+import static net.corda.training.TestUtils.BOB;
+import static net.corda.training.TestUtils.ALICE;
+import static net.corda.training.TestUtils.CHARLIE;
+
+/**
+ * Practical exercise instructions for Contracts Part 3.
+ * The objective here is to write some contract code that verifies a transaction to settle an [IOUState].
+ * Settling is more complicated than transfering and issuing as it requires you to use multiple state types in a
+ * transaction.
+ * As with the [IOUIssueTests] and [IOUTransferTests] uncomment each unit test and run them one at a time. Use the body
+ * of the tests and the task description to determine how to get the tests to pass.
+ */
+
+public class IOUSettleTests {
+
+    public interface Commands extends CommandData {
+	    class DummyCommand extends TypeOnlyCommandData implements Commands{}
+    }
+
+    static private final MockServices ledgerServices = new MockServices(Arrays.asList("net.corda.training"));
+
+	private Cash.State createCashState(AbstractParty owner, Amount<Currency> amount) {
+        OpaqueBytes defaultBytes = new OpaqueBytes(new byte[1]);
+	    PartyAndReference partyAndReference = new PartyAndReference(owner, defaultBytes);
+	    return new Cash.State(partyAndReference, amount, owner);
+    }
+
+    /**
+     * Task 1.
+     * We need to add another case to deal with settling in the [IOUContract.verify] function.
```

Comment:
Did you adopt these hints from Kotlin to Java?

Reply:
Modified tests in Settle (and double checked Issue / Transfer) to ensure that all tests instructions / hints were written for Java.